account_commission/models/account_move.py

```diff
@@ -21,6 +21,41 @@ class AccountMove(models.Model):
         compute="_compute_agents",
         search="_search_agents",
     )
+    settlement_count = fields.Integer(compute="_compute_settlement")
+    settlement_ids = fields.Many2many(
+        "commission.settlement",
+        string="Settlements",
+        compute="_compute_settlement",
+        copy=False,
+    )
+
+    def action_view_settlement(self):
+        settlements = self.mapped("settlement_ids")
+        action = self.env["ir.actions.actions"]._for_xml_id(
+            "commission.action_commission_settlement"
+        )
+        if len(settlements) > 1:
+            action["domain"] = [("id", "in", settlements.ids)]
+        elif len(settlements) == 1:
+            form_view = [(self.env.ref("commission.view_settlement_form").id, "form")]
+            if "views" in action:
+                action["views"] = form_view + [
+                    (state, view) for state, view in action["views"] if view != "form"
+                ]
+            else:
+                action["views"] = form_view
+            action["res_id"] = settlements.id
+        else:
+            action = {"type": "ir.actions.act_window_close"}
```

Comment:
No need of this. Simply don't populate any action in this case, but avoid this `else` that is not covered in tests.